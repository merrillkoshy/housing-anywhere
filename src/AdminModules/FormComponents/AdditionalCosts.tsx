import React, { useState } from "react";
import { Form, Col, Row } from "react-bootstrap";
import Switch from "react-switch";

const AdditionalCosts = () => {
	const [water, setWater] = useState(false);
	const [gas, setGas] = useState(false);
	const [electricity, setElectricity] = useState(false);
	const [internet, setInternet] = useState(false);

	return (
		<>
			<Form.Group as={Row} className="mb-3">
				<Form.Label as="legend" column>
					Water
				</Form.Label>
				<Col>
					<Switch
						id="water"
						onChange={() => {
							setWater(!water);
						}}
						checked={water}
					/>
				</Col>
			</Form.Group>
			<Form.Group as={Row} className="mb-3">
				<Form.Label as="legend" column>
					Gas
				</Form.Label>
				<Col>
					<Switch
						id="gas"
						onChange={() => {
							setGas(!gas);
						}}
						checked={gas}
					/>
				</Col>
			</Form.Group>
			<Form.Group as={Row} className="mb-3">
				<Form.Label as="legend" column>
					Electricity
				</Form.Label>
				<Col>
					<Switch
						id="electricity"
						onChange={() => {
							setElectricity(!electricity);
						}}
						checked={electricity}
					/>
				</Col>
			</Form.Group>
			<Form.Group as={Row} className="mb-3">
				<Form.Label as="legend" column>
					Internet
				</Form.Label>
				<Col>
					<Switch
						id="internet"
						onChange={() => {
							setInternet(!internet);
						}}
						checked={internet}
					/>
				</Col>
			</Form.Group>
		</>
	);
};
export default AdditionalCosts;
