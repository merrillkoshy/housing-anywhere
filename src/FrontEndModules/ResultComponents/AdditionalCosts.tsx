const AdditionalCosts = ({ costs }: { costs: any }) => {
	return (
		<details>
			<summary>
				<strong id="additionalCosts">Additional Costs</strong>
			</summary>
			<ul>
				<li id="water">Water:{costs?.water ? "✅" : "❌"}</li>
				<li id="gas">Gas:{costs?.gas ? "✅" : "❌"}</li>
				<li id="electricity">Electicity:{costs?.electricity ? "✅" : "❌"}</li>
				<li id="internet">Internet:{costs?.internet ? "✅" : "❌"}</li>
			</ul>
		</details>
	);
};
export default AdditionalCosts;
