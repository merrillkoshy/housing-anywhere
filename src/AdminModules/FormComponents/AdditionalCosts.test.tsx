import React from "react";
import { shallow } from "enzyme";
import AdditionalCosts from "./AdditionalCosts";

describe("Addl Costs module", () => {
	const AdditionalCostsModule = shallow(<AdditionalCosts />);
	it("renders", () => {
		expect(AdditionalCostsModule).toMatchSnapshot();
	});
	it("finds a water input which is a bool", () => {
		expect(AdditionalCostsModule.find("#water").exists()).toBe(true);
	});
	it("finds a gas input which is a bool", () => {
		expect(AdditionalCostsModule.find("#gas").exists()).toBe(true);
	});
	it("finds a electricity input which is a bool", () => {
		expect(AdditionalCostsModule.find("#electricity").exists()).toBe(true);
	});
	it("finds an internet input which is a bool", () => {
		expect(AdditionalCostsModule.find("#internet").exists()).toBe(true);
	});
});
