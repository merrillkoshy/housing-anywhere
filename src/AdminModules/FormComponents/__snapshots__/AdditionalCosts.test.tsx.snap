// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Addl Costs module renders 1`] = `
<Fragment>
  <FormGroup
    as={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": Object {
          "noGutters": false,
        },
        "render": [Function],
      }
    }
    className="mb-3"
  >
    <FormLabel
      as="legend"
      column={true}
      srOnly={false}
    >
      Water
    </FormLabel>
    <Col>
      <ReactSwitch
        activeBoxShadow="0 0 2px 3px #3bf"
        boxShadow={null}
        checked={false}
        checkedIcon={
          <svg
            height="100%"
            style={
              Object {
                "position": "absolute",
                "top": 0,
              }
            }
            viewBox="-2 -5 17 21"
            width="100%"
          >
            <path
              d="M11.264 0L5.26 6.004 2.103 2.847 0 4.95l5.26 5.26 8.108-8.107L11.264 0"
              fill="#fff"
              fillRule="evenodd"
            />
          </svg>
        }
        disabled={false}
        height={28}
        id="water"
        offColor="#888"
        offHandleColor="#fff"
        onChange={[Function]}
        onColor="#080"
        onHandleColor="#fff"
        uncheckedIcon={
          <svg
            height="100%"
            style={
              Object {
                "position": "absolute",
                "top": 0,
              }
            }
            viewBox="-2 -5 14 20"
            width="100%"
          >
            <path
              d="M9.9 2.12L7.78 0 4.95 2.828 2.12 0 0 2.12l2.83 2.83L0 7.776 2.123 9.9 4.95 7.07 7.78 9.9 9.9 7.776 7.072 4.95 9.9 2.12"
              fill="#fff"
              fillRule="evenodd"
            />
          </svg>
        }
        width={56}
      />
    </Col>
  </FormGroup>
  <FormGroup
    as={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": Object {
          "noGutters": false,
        },
        "render": [Function],
      }
    }
    className="mb-3"
  >
    <FormLabel
      as="legend"
      column={true}
      srOnly={false}
    >
      Gas
    </FormLabel>
    <Col>
      <ReactSwitch
        activeBoxShadow="0 0 2px 3px #3bf"
        boxShadow={null}
        checked={false}
        checkedIcon={
          <svg
            height="100%"
            style={
              Object {
                "position": "absolute",
                "top": 0,
              }
            }
            viewBox="-2 -5 17 21"
            width="100%"
          >
            <path
              d="M11.264 0L5.26 6.004 2.103 2.847 0 4.95l5.26 5.26 8.108-8.107L11.264 0"
              fill="#fff"
              fillRule="evenodd"
            />
          </svg>
        }
        disabled={false}
        height={28}
        id="gas"
        offColor="#888"
        offHandleColor="#fff"
        onChange={[Function]}
        onColor="#080"
        onHandleColor="#fff"
        uncheckedIcon={
          <svg
            height="100%"
            style={
              Object {
                "position": "absolute",
                "top": 0,
              }
            }
            viewBox="-2 -5 14 20"
            width="100%"
          >
            <path
              d="M9.9 2.12L7.78 0 4.95 2.828 2.12 0 0 2.12l2.83 2.83L0 7.776 2.123 9.9 4.95 7.07 7.78 9.9 9.9 7.776 7.072 4.95 9.9 2.12"
              fill="#fff"
              fillRule="evenodd"
            />
          </svg>
        }
        width={56}
      />
    </Col>
  </FormGroup>
  <FormGroup
    as={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": Object {
          "noGutters": false,
        },
        "render": [Function],
      }
    }
    className="mb-3"
  >
    <FormLabel
      as="legend"
      column={true}
      srOnly={false}
    >
      Electricity
    </FormLabel>
    <Col>
      <ReactSwitch
        activeBoxShadow="0 0 2px 3px #3bf"
        boxShadow={null}
        checked={false}
        checkedIcon={
          <svg
            height="100%"
            style={
              Object {
                "position": "absolute",
                "top": 0,
              }
            }
            viewBox="-2 -5 17 21"
            width="100%"
          >
            <path
              d="M11.264 0L5.26 6.004 2.103 2.847 0 4.95l5.26 5.26 8.108-8.107L11.264 0"
              fill="#fff"
              fillRule="evenodd"
            />
          </svg>
        }
        disabled={false}
        height={28}
        id="electricity"
        offColor="#888"
        offHandleColor="#fff"
        onChange={[Function]}
        onColor="#080"
        onHandleColor="#fff"
        uncheckedIcon={
          <svg
            height="100%"
            style={
              Object {
                "position": "absolute",
                "top": 0,
              }
            }
            viewBox="-2 -5 14 20"
            width="100%"
          >
            <path
              d="M9.9 2.12L7.78 0 4.95 2.828 2.12 0 0 2.12l2.83 2.83L0 7.776 2.123 9.9 4.95 7.07 7.78 9.9 9.9 7.776 7.072 4.95 9.9 2.12"
              fill="#fff"
              fillRule="evenodd"
            />
          </svg>
        }
        width={56}
      />
    </Col>
  </FormGroup>
  <FormGroup
    as={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": Object {
          "noGutters": false,
        },
        "render": [Function],
      }
    }
    className="mb-3"
  >
    <FormLabel
      as="legend"
      column={true}
      srOnly={false}
    >
      Internet
    </FormLabel>
    <Col>
      <ReactSwitch
        activeBoxShadow="0 0 2px 3px #3bf"
        boxShadow={null}
        checked={false}
        checkedIcon={
          <svg
            height="100%"
            style={
              Object {
                "position": "absolute",
                "top": 0,
              }
            }
            viewBox="-2 -5 17 21"
            width="100%"
          >
            <path
              d="M11.264 0L5.26 6.004 2.103 2.847 0 4.95l5.26 5.26 8.108-8.107L11.264 0"
              fill="#fff"
              fillRule="evenodd"
            />
          </svg>
        }
        disabled={false}
        height={28}
        id="internet"
        offColor="#888"
        offHandleColor="#fff"
        onChange={[Function]}
        onColor="#080"
        onHandleColor="#fff"
        uncheckedIcon={
          <svg
            height="100%"
            style={
              Object {
                "position": "absolute",
                "top": 0,
              }
            }
            viewBox="-2 -5 14 20"
            width="100%"
          >
            <path
              d="M9.9 2.12L7.78 0 4.95 2.828 2.12 0 0 2.12l2.83 2.83L0 7.776 2.123 9.9 4.95 7.07 7.78 9.9 9.9 7.776 7.072 4.95 9.9 2.12"
              fill="#fff"
              fillRule="evenodd"
            />
          </svg>
        }
        width={56}
      />
    </Col>
  </FormGroup>
</Fragment>
`;
